g2p/transducer/__init__.py

```diff
@@ -1249,3 +1254,53 @@ def check(self, tg: TransductionGraph, shallow=False, display_warnings=False):
                 else:
                     return False
         return result
+
+
+def preserve_case(
+    tg: TransductionGraph, case_equivalencies: Dict[str, str] = None
+) -> TransductionGraph:
+    if case_equivalencies is None:
+        case_equivalencies = {}
+    reverse_case_equivalencies = {v: k for k, v in case_equivalencies.items()}
+    all_lower_case_equivalencies = case_equivalencies.keys()
+    all_upper_case_equivalencies = case_equivalencies.values()
+    new_string = ""
+    for item in tg.substring_alignments():
+        in_sub = item[0]
+        out_sub = item[1]
+        any_in_upper = any(x.isupper() for x in in_sub)
+        any_in_lower = any(x.islower() for x in in_sub)
+        any_out_upper = any(x.isupper() for x in out_sub)
+        any_out_lower = any(x.islower() for x in out_sub)
+        # continue if character is un-caseable
+        if (
+            out_sub not in case_equivalencies
```

Comment:
So you're assuming case_equivalencies only lists complete rule outputs, rather than characters? What about `ʟ -> tλ` which would rely on the standard "t":"T" but would have to declare "λ":"⅄"?

Reply:
yes because we don't really have a notion of characters anywhere (re: the tokenization issue from before). here you would have to do `{'tλ': 'T⅄'}` I suppose